system/Helpers/cookie_helper.php

```diff
@@ -0,0 +1,155 @@
+ * THE SOFTWARE.
+ *
+ * @package    CodeIgniter
+ * @author     CodeIgniter Dev Team
+ * @copyright  Copyright (c) 2014 - 2016, British Columbia Institute of Technology (http://bcit.ca/)
+ * @license    http://opensource.org/licenses/MIT    MIT License
+ * @link       http://codeigniter.com
+ * @since      Version 3.0.0
+ * @filesource
+ */
+
+// --------------------------------------------------------------------
+
+/**
+ * CodeIgniter Cookie Helpers
+ *
+ * @package     CodeIgniter
+ * @subpackage  Helpers
+ * @category    Helpers
+ * @author      CodeIgniter Dev Team
+ * @link        http://codeigniter.com/user_guide/helpers/cookie_helper.html
+ */
+
+if ( ! function_exists('set_cookie'))
+{
+
+    /**
+     * Set cookie
+     *
+     * Accepts seven parameters, or you can submit an associative
+     * array in the first parameter containing all the values.
+     *
+     * @param   mixed   $name
+     * @param   string  $value    The value of the cookie
+     * @param   string  $expire   The number of seconds until expiration
+     * @param   string  $domain   For site-wide cookie. 
+     *                            Usually: .yourdomain.com
+     * @param   string  $path     The cookie path
+     * @param   string  $prefix   The cookie prefix
+     * @param   bool    $secure   true makes the cookie secure
+     * @param   bool    $httpOnly true makes the cookie accessible via 
+     *                            http(s) only (no javascript)
+     * @see     (\Config\Services::response())->setCookie()
+     * @see     \CodeIgniter\HTTP\Response::setCookie()
+     * @return  void
+     */
+    function set_cookie
+    (
+        $name, 
+        string $value    = '', 
+        string $expire   = '', 
+        string $domain   = '', 
+        string $path     = '/', 
+        string $prefix   = '', 
+        bool   $secure   = false, 
+        bool   $httpOnly = false
+    )
+    {
+        (\Config\Services::response(new \Config\App()))->setcookie
+        (
+            $name, 
+            $value, 
+            $expire, 
+            $domain, 
+            $path, 
+            $secure, 
+            $httpOnly
+        );
+    }
+
+}
+
+//--------------------------------------------------------------------
+
+if ( ! function_exists('get_cookie'))
+{
+
+    /**
+     * Fetch an item from the COOKIE array
+     *
+     * @param   mixed  $index
+     * @param   bool   $xssClean
+     * @see     (\Config\Services::request())->getCookie()
+     * @see     \CodeIgniter\HTTP\IncomingRequest::getCookie()
+     * @return  mixed
+     */
+    function get_cookie($index, bool $xssClean = false)
+    {
+        $app    = new \Config\App();
+        $prefix = $app->cookiePrefix;
+        $index  = $prefix . $index;
+        
+        $request = \Config\Services::request($app);
+        $filter  = true === $xssClean ? FILTER_SANITIZE_STRING : null;
+        $cookie  = $request->getCookie($index, $filter);
```

Comment:
If I'm not mistaken, `request->getCookie` handles adding the prefix already, doesn't it?